src/components/eRACommons.js

```diff
@@ -0,0 +1,212 @@
+import _ from 'lodash';
+import * as qs from 'query-string';
+import React from 'react';
+import { a, button, div, hh, label, span } from 'react-hyperscript-helpers';
+import { AuthenticateNIH, Researcher } from '../libs/ajax';
+import { Config } from '../libs/config';
+import { Storage } from '../libs/storage';
+
+
+export const eRACommons = hh(class eRACommons extends React.Component {
+
+  state = {
+    isAuthorized: false,
+    expirationCount: 0,
+    nihUsername: '',
+    nihError: false,
+    isHovered: false
+  };
+
+  componentDidMount = async () => {
+    if (this.props.location !== undefined && this.props.location.search !== '') {
+      this.authenticateAsNIHFCUser(this.props.location.search);
+    } else {
+      this.getResearcherProperties();
+    }
+  };
+
+  onMouseEnter = () => {
+    this.setState({ isHovered: true });
+  };
+
+  onMouseLeave = () => {
+    this.setState({ isHovered: false });
+  };
+
+  authenticateAsNIHFCUser = async (searchArg) => {
+    const currentUserId = Storage.getCurrentUser().dacUserId;
+    let isFcUser = this.verifyUser();
+    if (!isFcUser) {
+      Researcher.getPropertiesByResearcherId(currentUserId).then(
+        (response) => isFcUser = this.registerUserToFC(response),
+        () => this.setState({ nihError: true })
+      );
+    }
+    if (isFcUser) {
+      const parsedToken = qs.parse(searchArg);
+      this.verifyToken(parsedToken).then(
+        (decodedNihAccount) => {
+          AuthenticateNIH.saveNihUsr(decodedNihAccount).then(
+            () => this.getResearcherProperties(),
+            () => this.setState({ nihError: true })
+          );
+        },
+        () => this.setState({ nihError: true })
```

Comment:
The second fn on the `then` is for the reject case of the promise?
Why then is needed the catch? Can you explain this please?

Reply:
I don't see the catch in this method you're referring to. What line is that?

Reply:
Not the catch, sorry. The line 54 looks the same as 51.

Reply:
They're handling different error cases.
- Line 51 handles errors in `AuthenticateNIH.saveNihUsr`
- Line 54 handles errors in `this.verifyToken`

So, `verifyToken` can error out even if `saveNihUsr` succeeds, so we need to handle both cases.

Reply:
Yes! I see it now. Thanks Greg. :+1: